index.js

```diff
@@ -54,23 +54,20 @@ function serviceWorker (name, opts) {
             registration.onupdatefound = function () {
               var installingWorker = registration.installing
               installingWorker.onstatechange = function () {
-                switch (installingWorker.state) {
-                  case 'installed':
-                    if (navigator.serviceWorker.controller) {
-                      emitter.emit(events.UPDATED, registration)
-                    } else {
-                      emitter.emit(events.INSTALLED, registration)
-                    }
-                    break
-                  case 'redundant':
-                    emitter.emit(events.REDUNDANT, registration)
-                    break
+                if (installingWorker.state === 'installed') {
+                  if (navigator.serviceWorker.controller) {
+                    emitter.emit(events.UPDATED, registration)
+                  } else {
+                    emitter.emit(events.INSTALLED, registration)
+                  }
+                } else if (installingWorker.state === 'redundant') {
+                  emitter.emit(events.REDUNDANT, registration)
```

Comment:
bit of bikeshedding here; not a fan of switch statements haha — had trouble reading it; if/else is a bit more straight forward imo